src/main/java/com/github/minecraftschurlimods/arsmagicalegacy/common/affinity/AffinityHelper.java

```diff
@@ -115,7 +115,7 @@ public IAffinity getAffinityForStack(ItemStack stack) {
 
     @Override
     public double getAffinityDepth(Player player, ResourceLocation affinity) {
-        return getAffinityHolder(player).getAffinityDepth(affinity);
+        return player.isDeadOrDying() ? 0 : getAffinityHolder(player).getAffinityDepth(affinity);
```

Comment:
These checks should not be here they should occur in the call site

Reply:
This is more error-proof and reduces boilerplate at the call.

Reply:
then add another utility method but this one should not assume a 0 default